src/cdk/table/cell.ts

```diff
@@ -40,6 +40,14 @@ export class CdkColumnDef {
 
   /** @docs-private */
   @ContentChild(CdkHeaderCellDef) headerCell: CdkHeaderCellDef;
+
+  /**
+   * Returns the column's name with non-alphanumeric characters removed so it can be used
+   * as part of a CSS class. Allows for the special characters - and _.
+   */
+  getCssClassFriendlyName() {
+    return this.name.replace(/[^a-z0-9_-]/ig, '');
```

Comment:
I think you're doing more work than you need to by putting the logic here. I would do:

```ts
get name { return this._name; }
set name(value) {
  this._name = value;
  this.cssFriendlyName = (value | '').replace(/[^a-z0-9_-]/ig, '');
}
```